### ISSUE TITLE
Create absent directories when installing to flat files

### ISSUE THREAD

**Issue by ethanwhite** [weecology/retriever#1136]
The flat file engines will currently error if asked to install into a directory that doesn't exist, e.g.,

```
$ retriever install csv mammal-masses -t testdata/{db}_{table}.csv
=> Installing mammal-masses
[Errno 2] No such file or directory: 'testdata/mammal_masses_MammalMasses.csv'
Done!
```

We should create these directories on the fly. This may require some refactoring to split of the path into directories and file names, but I haven't looked yet.

**Comment by ashishpriyadarshiCIC** [comment]
Can I work on this?

**Comment by ethanwhite** [comment]
Thanks for working on this @ashishpriyadarshiCIC! Note that it is also related to #1137, so you and @harshitbansal05 may want to communicate with each other either here or on gitter.

**Comment by harshitbansal05** [comment]
@ashishpriyadarshiCIC are you working on it?

**Comment by ashishpriyadarshiCIC** [comment]
@harshitbansal05  Yes

**Comment by henrykironde** [comment]
Fixed in #1245
